CONTRIBUTING.md

```diff
@@ -6,7 +6,43 @@ Whether it's a bug report, new feature, correction, or additional documentation,
 
 Please read through this document before submitting any issues or pull requests to ensure we have all the necessary information to effectively respond to your bug report or contribution.
 
-## 1. 💪 Getting Started
+## 1. Developer Setup
+
+First install [anaconda](https://docs.anaconda.com/free/anaconda/install/windows/) and add to path by going to advanced system settings. Then, launch cmd.
+
+Ensure you have `make`. if it's not installed on your computer, Please follow this [article](https://earthly.dev/blog/makefiles-on-windows/).
+
+
+Use the following command to create a conda environment `discussion-agents` with Python version 3.10.13. Any Python version above 3.9 is viable.
+```
+conda create -n discussion-agents python=3.10.13
+```
+Now activate the environment.
+```
+conda activate discussion-agents
+```
+Next, we will install [Poetry](https://python-poetry.org/docs/) using [pipx](https://pipx.pypa.io/stable/docs/).
+```
+pip install pipx
+pipx install poetry
+```
+Make sure to add poetry to path by adding `C:\Users\<username>\.local\bin` to path in advanced system settings. For other operating systems, the path will be different. Ensure poetry is in the environment variable paths.
+
+Then clone the repository and enter the discussion-agents directory.
+``` 
+git clone https://github.com/alckasoc/discussion-agents/
+
+```
+Finally install all of the packages.
+```
+poetry install
+```
+### Pre-commit install and Pre-commit uninstall
+
+Pre-commit automatically runs specified checks before each commit to ensure code quality, adherence to best practices, and error prevention. If the checks fail, the commit is rejected until the issues are resolved. To set it up, execute pre-commit install. To remove it, use pre-commit uninstall.
+
```

Comment:
```suggestion
Pre-commit automatically runs specified checks before each commit to ensure code quality, adherence to best practices, and error prevention. If the checks fail, the commit is rejected until the issues are resolved. We use `pre-commit` to ensure code formatting on the developer's end, but often this may slow down development!

To uninstall `pre-commit,` run:

```
pre-commit uninstall
```

To install `pre-commit`, run:
```
pre-commit install
```
```